static/js/settings_org.js

```diff
@@ -404,28 +483,39 @@ function _set_up() {
         // grab the first alert available and use it for the status.
         var status = $("#admin-realm-notifications-stream-status");
 
-        var new_message_content_edit_limit_minutes = $("#id_realm_message_content_edit_limit_minutes").val();
         var new_allow_message_editing = $("#id_realm_allow_message_editing").prop("checked");
 
+        var new_message_edit_limit_unit = $('#edit_limit_unit_select').val();
+
+        var new_message_content_edit_limit = $("#id_realm_message_content_edit_limit").val();
+        var new_message_content_edit_limit_value
+            = parseInt(new_message_content_edit_limit, 10);
+
+        var new_message_content_edit_limit_seconds;
+
+        if (new_message_edit_limit_unit === "days") {
+            new_message_content_edit_limit_seconds
+                = new_message_content_edit_limit_value*(60*60*24);
+        } else if (new_message_edit_limit_unit === "hours") {
+            new_message_content_edit_limit_seconds = new_message_content_edit_limit_value*(60*60);
+        } else {
+            new_message_content_edit_limit_seconds = new_message_content_edit_limit_value*60;
+        }
         // If allow_message_editing is unchecked, message_content_edit_limit_minutes
         // is irrelevant.  Hence if allow_message_editing is unchecked, and
         // message_content_edit_limit_minutes is poorly formed, we set the latter to
         // a default value to prevent the server from returning an error.
```

Comment:
is this comment or the code under it still relevant? There should just be no way to save it with an invalid value